redshift/resource_redshift_datashare.go

```diff
@@ -0,0 +1,438 @@
+package redshift
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/lib/pq"
+)
+
+func redshiftDatashare() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+Defines a Redshift datashare. Datashares allows a Redshift cluster (the "consumer") to
+read data stored in another Redshift cluster (the "producer"). For more information, see
+https://docs.aws.amazon.com/redshift/latest/dg/datashare-overview.html
+
+The redshift_datashare resource should be defined on the producer cluster.
+
+Note: Data sharing is only supported on certain Redshift instance families,
+such as RA3.
+`,
+		Exists: RedshiftResourceExistsFunc(resourceRedshiftDatashareExists),
+		Create: RedshiftResourceFunc(resourceRedshiftDatashareCreate),
+		Read:   RedshiftResourceFunc(resourceRedshiftDatashareRead),
+		Update: RedshiftResourceFunc(resourceRedshiftDatashareUpdate),
+		Delete: RedshiftResourceFunc(resourceRedshiftDatashareDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
```

Comment:
Could you move the strings to consts as other resources have?

```
const (
  datashareNameAttr  = "name"
  datashareOwnerAttr = "owner"
  ...
)
```

In my opinion, this is a better approach rather than raw strings all around, it eliminates the possibility of a typo and makes it easier to rename an attribute in the future if needed.

Reply:
@winglot - I've moved the attribute names to constants and amended the commit.

Reply:
Rebased #20 on these changes as well.